server/bin/pbench-index

```diff
@@ -1,259 +1,3563 @@
+                    first = tstamp
+                elif first != tstamp:
+                    self.logger.warning("tool-data-indexing: {} csv files have"
+                            " inconsistent timestamps per row", self.toolname)
+                    self.counters['inconsistent_timestamps_across_csv_files'] += 1
+                    break
+            # We are now ready to create a base document per identifier to
+            # hold all the fields from the various columns. Given the two
+            # input dictionaries, "identifiers" and "metadata", we create
+            # an output dictionary, "datum", which has keys for all the
+            # identifiers and a base dictionary for forming the JSON docs.
+
+            # For example, given these inputs:
+            #   * identifiers = { "id0": True, "id1": True }
+            #   * metadata = { "id0": { "f1": "foo", "f2": "bar" },
+            #                  "id1": { "f1": "faz", "f2": "baz" } }
+            # The for loop below would generate the following dictionary:
+            #   * datum = { "id0": { "@timestamp": ts_str,
+            #                        "run": self.run_metadata,
+            #                        "sample": self.sample_metadata,
+            #                        "iteration": self.iteration_metadata,
+            #                        self.toolname: { "id": "id0",
+            #                                         "f1": "foo",
+            #                                         "f2": "bar" } },
+            #               "id1": { "@timestamp": ts_str,
+            #                        "run": self.run_metadata,
+            #                        "sample": self.sample_metadata,
+            #                        "iteration": self.iteration_metadata,
+            #                        self.toolname: { "id": "id1",
+            #                                         "f1": "faz",
+            #                                         "f2": "baz" } },
+
+            # The timestamp is taken from the "first" timestamp, converted
+            # to a floating point value in seconds, and then formatted as a
+            # string.
+            if prev_ts is not None:
+                assert prev_ts <= first, "prev_ts %r > first %r" % (prev_ts, first)
+            ts_val = self.mk_abs_timestamp_millis(first)
+            prev_ts = first
+            datum = _dict_const()
+            for identifier in identifiers.keys():
+                datum[identifier] = _dict_const([
+                    # Since they are all the same, we use the first to
+                    # generate the real timestamp.
+                    ('@timestamp', ts_val),
+                    ('@timestamp_original', str(first)),
+                    ('run', self.run_metadata),
+                    ('iteration', self.iteration_metadata),
+                    ('sample', self.sample_metadata),
+                    (self.toolname, _dict_const())
+    
```

Comment:
Do we need a signal handler to be able to do this deletion (and maybe others) if the process gets killed?

Reply:
Are you referring to the `os.remove(ie_filename)` at line 3563?